### DISCUSSION TITLE
How to do with existing data model?

### DISCUSSION THREAD

**Discussion by dcheung2** [Netflix/dgs-framework#30] [General]
If I understand it correctly, DGS is a schema first approach, unlike code first approach SPQR.
https://netflix.github.io/dgs/generating-code-from-schema/

But what if there is a model out there I'd like to reuse?

I mean, for example, I want to use DGS in a new orchestration layer,  which I may want to reuse data-model 
- in a common model JAR exported from a upstream using SPQR project.
- in DTO generated from ORM or JooQ

If the code mismatch from the graphql schema file, can I I'd expect some sort of compile time error??

**Comment by paulbakker** [comment]
There are a few things to consider in this scenario:

- Code gen is optional. If your existing model fits well, just use it. However, as a best practice, always put your schema design first and don’t just “data dump” what you already have.
- You can add extra types at runtime using this: https://netflix.github.io/dgs/advanced/schema-from-code/. We use this at Netflix for some use cases where the GraphQL API is generated from another schema/model. We don’t recommend it for most use cases, but it can be very useful.

That being said, their is no functionality to verify your existing model against a schema.
